tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenClientTest.cs

```diff
@@ -0,0 +1,217 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public abstract class JsonDrivenClientTest
```

Comment:
This is the base class of all JSON driven tests (for now).

There will probably be one more base class above this one that doesn't have a _mongoClient. It would be used for the BSON corpus tests, which don't need a MongoDB server (as far as I know).